src/router/STBRouter.sol

```diff
@@ -0,0 +1,116 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity ^0.8.20;
+
+import {L1YearnEscrow} from "../L1YearnEscrow.sol";
+import {L1Deployer} from "../L1Deployer.sol";
+
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import {Multicall} from "./Multicall.sol";
+import {PeripheryPayments} from "./PeripheryPayments.sol";
+
+import {IPermit2} from "./IPermit2.sol";
+
+contract STBRouter is Multicall, PeripheryPayments {
```

Comment:
I'm feeling dumb ... isn't PeripheryPayments completely open to all callers?

Reply:
yes. Any tokens left in the router after a txn can get pulled by anyone.

The idea is the multicall allows any of them to be used through a delegatecall

Reply:
do you know if the rationale for Uniswap was to not keep internal accounting, and thus simplify the implementation?